k8s/predelete.go

```diff
@@ -0,0 +1,144 @@
+package k8s
+
+import (
+	"context"
+	"tyk/tyk/bootstrap/pkg/constants"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ExecutePreDeleteOperations executes operations needed in pre-delete chart hook one by one.
+func (c *Client) ExecutePreDeleteOperations() error {
+	c.l.Info("Running pre delete hook")
+
+	if err := c.deleteOperatorSecret(); err != nil {
+		return err
+	}
+
+	if err := c.deletePortalSecret(); err != nil {
+		return err
+	}
+
+	if err := c.deleteBootstrappingJobs(); err != nil {
+		return err
+	}
+
+	c.l.Info("Run pre delete hook successfully")
+
+	return nil
+}
+
+// deleteOperatorSecret deletes the Kubernetes secret created specifically for Tyk Operator.
+func (c *Client) deleteOperatorSecret() error {
+	secrets, err := c.clientSet.
+		CoreV1().
+		Secrets(c.appArgs.K8s.ReleaseNamespace).
+		List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	found := false
+
+	for i := range secrets.Items {
+		value := secrets.Items[i]
+		if value.Name == c.appArgs.OperatorKubernetesSecretName {
+			err = c.clientSet.
+				CoreV1().
+				Secrets(c.appArgs.K8s.ReleaseNamespace).
+				Delete(context.TODO(), value.Name, metav1.DeleteOptions{})
+
+			if err != nil {
+				return err
+			}
+
+			found = true
+
+			break
+		}
+	}
+
+	if !found {
+		c.l.Info("A previously created operator secret has not been identified")
+	} else {
+		c.l.Info("A previously created operator secret was identified and deleted")
+	}
+
+	return nil
+}
+
+// deletePortalSecret deletes the Kubernetes secret created specifically for Tyk Developer Portal.
+func (c *Client) deletePortalSecret() error {
```

Comment:
Since deleteOperatorSecret and deletePortalSecret is doing the same thing just difference is in the secret name should we create a function that takes secretName as parameter?

Reply:
makes sense 👍 will adapt this approach